include/libdnf/rpm/package_set.hpp

```diff
@@ -70,45 +69,98 @@ class PackageSet {
+
+    /// Set difference: elements in the current set that are not in the `other` set.
+    ///
+    /// @param other The set to check for differences in.
+    /// @exception UsedDifferentSack When the sets entering the operation do not share the same PackageSack.
+    /// @since 5.0
+    //
+    // @replaces libdnf/hy-query.h:function:hy_query_difference(HyQuery q, HyQuery other)
+    // @replaces libdnf/sack/query.hpp:method:queryDifference(Query & other)
+    void difference(const PackageSet & other) { *this -= other; }
+
+    /// Remove all packages from the set.
+    ///
+    /// @since 5.0
+    //
+    // @replaces libdnf:sack/packageset.hpp:method:PackageSet.clear()
     void clear() noexcept;
 
-    /// @replaces libdnf:sack/packageset.hpp:method:PackageSet.empty()
+    /// @return `true` if the set is empty, `false` otherwise.
+    /// @since 5.0
+    //
+    // @replaces libdnf:sack/packageset.hpp:method:PackageSet.empty()
     bool empty() const noexcept;
 
-    /// @replaces libdnf:sack/packageset.hpp:method:PackageSet.set(DnfPackage * pkg)
-    /// @replaces libdnf:hy-packageset.h:function:dnf_packageset_add(DnfPackageSet * pset, DnfPackage * pkg)
+    /// Add `pkg` to the set.
+    ///
+    /// @param pkg Package to be added to the set.
+    /// @since 5.0
+    //
+    // @replaces libdnf:sack/packageset.hpp:method:PackageSet.set(DnfPackage * pkg)
+    // @replaces libdnf:hy-packageset.h:function:dnf_packageset_add(DnfPackageSet * pset, DnfPackage * pkg)
     void add(const Package & pkg);
 
-    /// @replaces libdnf:sack/packageset.hpp:method:PackageSet.has(DnfPackage * pkg)
+    /// @return `true` if a package is in the set, `false` otherwise.
+    /// @param pkg Package that is tested for presence.
+    /// @since 5.0
+    //
+    // @replaces libdnf:sack/packageset.hpp:method:PackageSet.has(DnfPackage * pkg)
     bool contains(const Package & pkg) const noexcept;
 
+    /// Remove `pkg` from the set.
+    ///
+    /// @param pkg Package to be removed from the set.
+    /// @since 5.0
     void remove(const Package & pkg);
 
-    /// @replaces libdnf:sack/packageset.hpp:method:PackageSet.getSack()
+    // TODO(dmach): replace with get_base()?
+    // @replaces libdnf:sack/packageset.hpp:method:PackageSet.getSack()
     PackageSackWeakPtr get_sack() const;
 
-    /// @replaces libdnf:sack/packageset.hpp:method:PackageSet.size()
-    /// @replaces libdnf:hy-packageset.h:function:dnf_packageset_count(DnfPackageSet * pset)
+    /// @return Number of elements in the set.
+    //
+    // @replaces libdnf:sack/packageset.hpp:method:PackageSet.size()
+    // @replaces libdnf:hy-packageset.h:function:dnf_packageset_count(DnfPackageSet * pset)
     size_t size() const noexcept;
 
     void swap(PackageSet & other) noexcept;
 
 private:
-    friend PackageSetIterator;
-    friend PackageQuery;
-    friend Transaction;
-    friend libdnf::Goal;
-    friend libdnf::Swdb;
+    friend class PackageSetIterator;
+    friend class PackageQuery;
```

Comment:
I propose to change `class PackageQuery;` to `struct PackageQuery;`. Anyway we have to change it for C bindings.
I suggest to keep`friend PackageQuery;` is a better idea.

Reply:
As discussed in the meeting earlier today, I'd prefer to stick with `class` and if we decide to support C API, then mass-rename `class` to `struct` everywhere in the code. AFAIK even `friend struct <Class>` should work fine and is straightforward to replace.

Reply:
Is there a reason we can't just have SWIG generate C bindings?